Modify rule S6223: Add missing "Noncompliant Code Example" (#217)
SonarSource/rspec: rules/S6223/cfamily/rule.adoc

The ``++range++`` counterparts of the above algorithms (in namespace ``++std::ranges++``) return a ``++subrange++`` of remaining elements, but otherwise, they exhibit the same behavior.

The rule raises an issue when the result of the invocation of ``++remove++``, ``++remove_if++`` or ``++unique++`` from namespace ``++std++`` or ``++std::ranges++`` is ignored.

== Noncompliant Code Example

----
void func(std::vector<int>& v) {
  std::remove_if(v.begin(), v.end(), [](int x) { return x % 2 == 0; }); // Noncompliant
}

auto largestFiles(vector<char*> fileNames, int n) {
  remove(fileNames.begin(), fileNames.end(), nullptr); // Noncompliant
  return max_element(fileNames.begin(), fileNames.end(),
    [](auto f1, auto f2) {

>       return filesystem::file_size(f1) < filesystem::file_size(f2);}); // Undefined behavior: dereferencing null pointers

Reviewer: ```suggestion
      // Undefined behavior: accessing unspecifed elements and potentially dereferencing null pointers
      return filesystem::file_size(f1) < filesystem::file_size(f2);});
```

}

auto largestFiles(vector<char*> fileNames, int n) {

>   auto it = remove(fileNames.begin(), fileNames.end(), nullptr); // Compliant, even if the return value is not used in the erase-remove idiom.

Reviewer: ```suggestion
  // Compliant, the value is used (without the erase-remove idiom, it did not apply here)
  auto it = remove(fileNames.begin(), fileNames.end(), nullptr);
```
